gr-blocks/lib/blockinterleaver_xx_impl.cc

```diff
@@ -0,0 +1,178 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2022 Johannes Demel.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "blockinterleaver_xx_impl.h"
+#include <gnuradio/io_signature.h>
+#include <type_traits>
+
+namespace gr {
+namespace blocks {
+
+template <class T>
+typename blockinterleaver_xx<T>::sptr blockinterleaver_xx<T>::make(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed)
+{
+    return gnuradio::make_block_sptr<blockinterleaver_xx_impl<T>>(
+        interleaver_indices, interleave_mode, is_packed);
+}
+
+
+/*
+ * These are explicit template declarations to make sure that all these specialized make
+ * functions are present in the compiled shared library. So far this is unnecessary for
+ * other blocks. However, the reasons are yet unknown.
+ */
+template typename blockinterleaver_xx<uint8_t>::sptr blockinterleaver_xx<uint8_t>::make(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed);
+
+template typename blockinterleaver_xx<gr_complex>::sptr
+blockinterleaver_xx<gr_complex>::make(std::vector<size_t> interleaver_indices,
+                                      bool interleave_mode,
+                                      bool is_packed);
+
+template typename blockinterleaver_xx<float>::sptr blockinterleaver_xx<float>::make(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed);
+
+template typename blockinterleaver_xx<int32_t>::sptr blockinterleaver_xx<int32_t>::make(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed);
+
+template typename blockinterleaver_xx<int16_t>::sptr blockinterleaver_xx<int16_t>::make(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed);
+
+
+template <class T>
+blockinterleaver_xx_impl<T>::blockinterleaver_xx_impl(
+    std::vector<size_t> interleaver_indices, bool interleave_mode, bool is_packed)
+    : gr::sync_block("blockinterleaver",
+                     gr::io_signature::make(1, 1, sizeof(T)),
+                     gr::io_signature::make(1, 1, sizeof(T))),
+      d_interleave_mode(interleave_mode),
+      d_is_packed(is_packed),
+      d_unpacker(std::make_unique<gr::blocks::kernel::unpack_k_bits>(8)),
+      d_packer(std::make_unique<gr::blocks::kernel::pack_k_bits>(8)),
+      d_interleaver(std::make_unique<block_interleaving>(interleaver_indices))
+{
+    if (not std::is_same<T, uint8_t>::value) {
```

Comment:
This will hopefully fix the broken Windows build.
```suggestion
    if (!std::is_same<T, uint8_t>::value) {
```

Reply:
Thanks! I missed this one. I was searching for `or` and `and`.